components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.cache.ui/src/main/resources/web/cache-mediator/update-mediator.jsp

```diff
@@ -55,39 +51,55 @@
             cacheMediator.setCollector(false);
         }
     }
+
     if (notNullChecker(cacheTimeout)) {
         try {
             cacheMediator.setTimeout(Long.parseLong(cacheTimeout));
         } catch (NumberFormatException e) {
 
         }
+    } else {
+        cacheMediator.setTimeout(CachingConstants.DEFAULT_TIMEOUT);
     }
+
     if (notNullChecker(maxMsgSize)) {
         try {
             cacheMediator.setMaxMessageSize(Integer.parseInt(maxMsgSize));
         } catch (NumberFormatException e) {
 
```

Comment:
Shouldn't we throw an error in the case of a NumberFormatException?

Reply:
This is handled in the UI validation in mediator-util.js . I have added a comment to reflect that